### PR TITLE
Edit Report Name flow

### PR DESCRIPTION
**Pull request by yksolanki9**
https://user-images.githubusercontent.com/39493102/134898865-4d1da938-f6a8-4d48-9c5b-02406c3e7776.mp4

### CODE REVIEW

**Comment by Dimple16** [comment]
few comments:
- The icon color should be `$primary`
- Let us add a left margin to the icon temporarily - It looks very near to the name
- Edit name header in the pop up is not center-aligned
- Aren't we showing an error message if the report name is erased completely?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
resolve conflict

**Review by Dimple16 — APPROVED** [review]

File: src/app/core/services/report.service.ts [hunk 1]
```diff
@@ -198,6 +198,16 @@ export class ReportService {
...
   }
 
+  @CacheBuster({
+    cacheBusterNotifier: reportsCacheBuster$,
+  })
+  updateReportDetails(erpt) {
+    const reportData = this.dataTransformService.unflatten(erpt);
+    return this.apiService
+      .post('/reports', reportData.rp)
+      .pipe(switchMap((res) => this.clearTransactionCache().pipe(map(() => res))));
```
Comment (Julias0) [inline src/app/core/services/report.service.ts:208]: this.clearTransactionCache() - do this inside a tap

File: src/app/fyle/my-view-report/my-view-report.page.ts [hunk 2]
```diff
@@ -167,6 +172,33 @@ export class MyViewReportPage implements OnInit {
...
+      },
+      cssClass: 'fy-dialog-popover',
+    });
+
+    await editReportNamePopover.present();
+    const { data } = await editReportNamePopover.onWillDismiss();
+
+    if (data && data.reportName) {
+      erpt.rp_purpose = data.reportName;
+      from(this.loaderService.showLoader())
```
Comment (Julias0) [inline src/app/fyle/my-view-report/my-view-report.page.ts:190]: extract the content inside the if into a function

**Review by Julias0 — APPROVED** [review]

**Pull request merged**